src/main/java/com/denizenscript/clientizen/objects/LocationTag.java

```diff
@@ -0,0 +1,203 @@
+            // If 3 values, worldless location format: x,y,z
+            // If 4 values, standard dScript location format: x,y,z,world
+            // If 5 values, worldless location with pitch/yaw: x,y,z,pitch,yaw
+            // If 6 values, location with pitch/yaw: x,y,z,pitch,yaw,world
+            double x = Double.parseDouble(split.get(0));
+            double y = Double.parseDouble(split.get(1));
+            double z = 0;
+            float yaw = 0, pitch = 0;
+            String world = null;
+            if (size > 2) {
+                z = Double.parseDouble(split.get(2));
+            }
+            if (size == 5 || size == 6) {
+                pitch = Float.parseFloat(split.get(3));
+                yaw = Float.parseFloat(split.get(4));
+            }
+            if (size == 4 || size == 6) {
+                world = split.get(size - 1);
+                if (world.startsWith("w@")) {
+                    world = world.substring("w@".length());
+                }
+            }
+            return new LocationTag(x, y, z, yaw, pitch).setWorld(world);
+        }
+        catch (Exception e) {
+            if (context == null || context.showErrors()) {
+                Debug.echoError("valueOf LocationTag returning null: " + string + " (internal exception:" + e.getMessage() + ")");
+            }
+            return null;
+        }
+    }
+
+    public static boolean matches(String string) {
+        if (string.startsWith("l@")) {
+            return true;
+        }
+        return valueOf(string, CoreUtilities.noDebugContext) != null;
+    }
+
+    public int getBlockX() {
+        return MathHelper.floor(x);
+    }
+    public int getBlockY() {
+        return MathHelper.floor(y);
+    }
+
+    public int getBlockZ() {
+        return MathHelper.floor(z);
+    }
+
+    public static void register() {
+        tagProcessor.registerStaticTag(ElementTag.class, "x", (attribute, object) -> {
+            return new ElementTag(object.x);
+        });
+        tagProcessor.registerStaticTag(ElementTag.class, "y", (attribute, object) -> {
+            return new ElementTag(object.y);
+        });
+        tagProcessor.registerStaticTag(ElementTag.class, "z", (attribute, object) -> {
+            return new ElementTag(object.z);
+        });
+        tagProcessor.registerStaticTag(ElementTag.class, "yaw", (attribute, object) -> {
+            return new ElementTag(object.yaw);
+        });
+        tagProcessor.registerStaticTag(ElementTag.class, "pitch", (attribute, object) -> {
+            return new ElementTag(object.pitch);
+        });
+        tagProcessor.registerStaticTag(ElementTag.class, "world", (attribute, object) -> {
+            return new ElementTag(object.world);
+        });
+        tagProcessor.registerStaticTag(LocationTag.class, "block", (attribute, object) -> {
+            return new LocationTag(object.getBlockX(), object.getBlockY(), object.getBlockZ()).setWorld(object.world);
+        });
```

Comment:
Should these preemptively be doc'd going forward?

Reply:
~~Stole~~ added some docs